### PR TITLE
Begin deprecation of `green_or_brown`

### PR DESCRIPTION
**Pull request by jdhoffa**
This PR begins the process of deprecating the `green_or_brown` dataset, in favour of `increasing_or_decreasing`. 

Note: that these datasets are not identical stand-ins, since the names of the columns and also entries within the dataset also necessitated a change, however, the content remains conceptually identical. 

We will offer one release on CRAN with these superseded datasets, before hard deprecating them. 

Relates to #307

### CODE REVIEW

**Comment by jdhoffa** [comment]
``` r
library(r2dii.data)

# deprecated dataset warns
head(green_or_brown)
#> Warning: The dataset `green_or_brown`is superseded as of r2dii.data 0.3.2.
#> Please use `increasing_or_decreasing` instead.
#>       sector    technology green_or_brown
#> 1 automotive      electric          green
#> 2 automotive        hybrid          green
#> 3 automotive           ice          brown
#> 4 automotive      fuelcell          green
#> 5      power      hydrocap          green
#> 6      power renewablescap          green

# new dataset, with new content
head(increasing_or_decreasing)
#>       sector    technology increasing_or_decreasing
#> 1 automotive      electric               increasing
#> 2 automotive        hybrid               increasing
#> 3 automotive           ice               decreasing
#> 4 automotive      fuelcell               increasing
#> 5      power      hydrocap               increasing
#> 6      power renewablescap               increasing
```

<sup>Created on 2023-06-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: data/green_or_brown.R
Comment (maurolepore) [inline data/green_or_brown.R:1]: :sunglasses:

File: data-raw/data_dictionary/increasing_or_decreasing.csv
Comment (cjyetman) [inline data-raw/data_dictionary/increasing_or_decreasing.csv:1]: two files with the name `increasing_or_decreasing.csv` in two separate directories with different content seems confusing

File: man/green_or_brown.Rd
Comment (cjyetman) [inline man/green_or_brown.Rd:1]: I might not fully understand the deprecation/superseding process, but I would expect to see this replaced with an equivalent for the new `increasing_or_decreasing` object

File: data-raw/data_dictionary/increasing_or_decreasing.csv
Comment (jdhoffa) [reply data-raw/data_dictionary/increasing_or_decreasing.csv:1]: This is in line with the rest of the package strategy, for template datasets. 
The directory `data-raw` contains any necessary raw data input files used to derive the exported datasets of the package. 

The directory `data-dictionary` contains the complete data dictionary for each exported dataset. To make it easier to manage, each dictionary element is separated by the dataset it defines, and named after it

**Pull request merged**